src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -1212,17 +1212,26 @@ private void registerKeyStoreProviderOnConnection(String keyStoreAuth, String ke
             String keyStoreLocation) throws SQLServerException {
         if (null == keyStoreAuth) {
             // secret and location must be null too.
-            if ((null != keyStoreSecret)) {
-                MessageFormat form = new MessageFormat(
-                        SQLServerException.getErrString("R_keyStoreAuthenticationNotSet"));
-                Object[] msgArgs = {"keyStoreSecret"};
-                throw new SQLServerException(form.format(msgArgs), null);
-            }
-            if (null != keyStoreLocation) {
-                MessageFormat form = new MessageFormat(
-                        SQLServerException.getErrString("R_keyStoreAuthenticationNotSet"));
-                Object[] msgArgs = {"keyStoreLocation"};
-                throw new SQLServerException(form.format(msgArgs), null);
+            // EXCEPT if we are trying to decrypt from a non windows machine. We will allow it
+            if (null != keyStoreSecret && null != keyStoreLocation) {
+                SQLServerColumnEncryptionCertificateStoreProvider provider =
+                        new SQLServerColumnEncryptionCertificateStoreProvider();
+                provider.setKeyStorePath(keyStoreLocation);
+                provider.setKeyStorePwd(keyStoreSecret);
+                systemColumnEncryptionKeyStoreProvider.put(provider.getName(), provider);
+            } else {
+                if ((null != keyStoreSecret)) {
+                    MessageFormat form = new MessageFormat(
+                            SQLServerException.getErrString("R_keyStoreAuthenticationNotSet"));
+                    Object[] msgArgs = {"keyStoreSecret"};
+                    throw new SQLServerException(form.format(msgArgs), null);
+                }
+                if (null != keyStoreLocation) {
+                    MessageFormat form = new MessageFormat(
+                            SQLServerException.getErrString("R_keyStoreAuthenticationNotSet"));
```

Comment:
**Reporter**: CHECKSTYLE
**Rule**: com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck
**Severity**: ERROR
**File**: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java L1231

Line is longer than 80 characters (found 95).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<289607437>*